### ISSUE TITLE
Using orca I can't browse by word in a commit message

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#95326]
Issue Type: <b>Bug</b>

1. Launch orca.
2. Launch vscode using a folder inside a git repository.
3. Press ctrl+shift+g.
4. Type a message in the commit message editor.
5. Press ctrl+home.
6. Explore the message using ctrl+left and ctrl+wright.

Orca reads something like blank or the intire message.

CC @isidorn 

VS Code version: Code - Insiders 1.45.0-insider (1c36d747fbf113144d3bf82b0aa3cbd290714e38, 2020-04-15T05:37:14.972Z)
OS version: Linux x64 5.6.4-arch1-1

**Comment by isidorn** [comment]
@jvesouza thanks for filling this issue. However I could not reproduce it. When I follow your steps and navigate by word Orca reads the first letter of a word. Do I need to have Orca somehow configured?
The commit message editor is just like the regular editor for files, so first of all does this ctrl+left, ctrl+right navigation work for you in the regular editor?

fyi @joanmarie

**Comment by jvesouza** [comment]
@isidorn  Did you press the enter key after typing the commit message? It seems that the problem only happens when the message has only one line and we don't press the enter key when we finish typing it. I just found out that this problem doesn't just happen when we're typing a commit message, it happens when we create a new file too.

steps to reproduce:
1. Launch orca.
2. Create a new file in vscode.
3. type the following text without press the enter key in the end.
this is a test.

4. Press ctrl+home and use ctrl+left and ctrl+right. 
Orca do not read the words this is a test

**Comment by isidorn** [comment]
Thanks but works for me also when I have everything in one line. Can you repro if you also do not press `ctrl+home`? Since I do not press that keybinding since I do not have it on my keyboard.

**Comment by jvesouza** [comment]
@isidorn  After press this is a 'test' without the quotations, the following happens if I try browse using ctrl+left:
* orca reads 'test' when I press ctrl+left for the first time.
* orca reads 'a test' when I press ctrl+left for the second time.
* orca reads 'is a test' when I press ctrl+left for the third time.
* orca reads 'this is a test' when I press ctrl+left for the fourth time.

**Comment by joanmarie** [comment]
@jvesouza I cannot reproduce this problem either. Mind emailing me a full debug.out so I can see what might be going on?

**Comment by jvesouza** [comment]
@joanmarie  I sent a debug file, please let me know if you haven't received it.
Thanks.

**Comment by joanmarie** [comment]
@jvesouza Just took a look at the new one you sent me. Thanks!

I see all sorts of brokenness coming from Orca's call to pyatspi's

     getTextAtOffset(offset, pyatspi.TEXT_BOUNDARY_WORD_START)

Here's a not-broken example from your output:
> Word at offset 10 is 'test' (10-14)

But then look at the ones which follow:
> Word at offset 5 is 'is a test' (5-0)
> Word at offset 8 is 'a test' (8-0)
> Word at offset 0 is 'this is a test' (0-14)

Compare that with mine (not broken):
> Word at offset 10 is 'test' (10-14)
> Word at offset 8 is 'a ' (8-10)
> Word at offset 5 is 'is ' (5-8)
> Word at offset 0 is 'This ' (0-5)

So this is definitely not an Orca bug. It's likely not a VSCode bug either. The AtkText implementation is in Chromium. But it's hard to file the Chromium bug without being able to reproduce it. :frowning: 

Do you see this problem with `textarea` elements in Chrome/Chromium?

**Comment by jvesouza** [comment]
@joanmarie @isidorn Could this problem be related to the electron version?
The electron used in my build is 7.2.1 and the chrome version is 78.0.3904.130

**Comment by joanmarie** [comment]
Ohhhh man. 78.0.x. My guess would be yeah. Chromium wasn't even remotely usable until like 80.0.x.

@isidorn The docs related to VSCode + Orca should probably address this.

**Comment by jvesouza** [comment]
I am running code-insiders downloaded from microsoft.
Overall it has worked quite well.
I am not sure if I can build using a different chrome version.
To be honest I don't know how to build VSCode!

**Comment by joanmarie** [comment]
@jvesouza I don't see the problem with code-insiders either. And now that I look at the version, it's the same thing you have. So.... I'm honestly pretty perplexed as to why you're having this problem but I'm not. Maybe @isidorn or one of his colleagues has some insight?

**Comment by isidorn** [comment]
Thanks for looking into this.
Yes this sounds like some Chrome fishiness.

It is not so trivial to build VS Code out of source with a later Electron version. However luckily thanks to @deepak1556 we have VS Code exploration builds that run on Chrome 80.
@jvesouza can you maybe give those a spin just so we know if this one gets fixed once we update to latest Electron version.
@deepak1556 speaking of that do we plan to update to Electron 8 this milestone or next one?

Here are the download links for the exploration build
* https://az764295.vo.msecnd.net/exploration/d45720441c97c3cca64c2868e2399c07a207f7e4/code-exploration-1586930354.tar.gz
* https://az764295.vo.msecnd.net/exploration/d45720441c97c3cca64c2868e2399c07a207f7e4/code-exploration_1.45.0-1586929972_amd64.deb
* https://az764295.vo.msecnd.net/exploration/2766badcb6c29445f5289033f1343f373802315c/code-exploration-1.44.0-1585360385.el7.x86_64.rpm
* https://az764295.vo.msecnd.net/exploration/2766badcb6c29445f5289033f1343f373802315c/code-exploration-1585360850.snap

**Comment by deepak1556** [comment]
@isidorn we won't go for Electron 8 in this milestone, its hard to say about next milestone as the priority lies with the node-free renderer work which would allow us skip versions and jump directly to Electron 10. 

But we will be setting up nightly exploration builds in this milestone, to keep Electron 8 alive.

**Comment by jvesouza** [comment]
@isidorn I did a quick test using a suggested version and the problem seems to be gone.
The version uses electron 8.2.2 and I only got accessibility when I used the --force-renderer-accessibility flag explicitly.
Without this flag no accessibility.
I believe that this issue can be closed.
Thank you all and forgive me for the noise!

**Comment by isidorn** [comment]
@jvesouza it is not noise, but an issue with Chrome. However since we do not have steps and seems to be fixed in latest I prefer to close this one. As a workaround you can use the exploration build. 

@deepak1556 sounds great, thanks for letting me know!
